src/Commands/GenerateSitemap.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace Rapidez\Statamic\Commands;
+
+use Illuminate\Console\Command;
+use Statamic\Facades\Collection as CollectionFacade;
+use Statamic\Facades\Site as SiteFacade;
+use Statamic\Sites\Site;
+use Statamic\Entries\Collection;
+use Statamic\Eloquent\Entries\Entry;
+use Statamic\Facades\Entry as EntryFacade;
+use Statamic\Facades\Taxonomy as TaxonomyFacade;
+use Statamic\Taxonomies\Taxonomy;
+use Statamic\Taxonomies\LocalizedTerm;
+
+class GenerateSitemap extends Command
+{
+    protected $signature = 'rapidez:statamic:generate:sitemap';
+
+    protected $description = 'Generate a sitemap based on Statamic collections & taxonomies';
+
+
+    public function handle()
+    {
+        foreach (SiteFacade::all() as $site) {
+            $this->generateStoreSitemap($site);
+        }
+
+        return Command::SUCCESS;
+    }
+
+    protected function generateStoreSitemap(Site $site) : void
+    {
+        $sitemap = '<?xml version="1.0" encoding="UTF-8"?>';
+        $sitemap .= '<sitemapindex xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">';
+
+        $collections = CollectionFacade::all();
+        $collections = $collections->filter(fn (Collection $collection) => $collection->route($site->handle()));
+
+        foreach ($collections as $collection) {
+            /* @var Collection $collection */
+            if ($collection->queryEntries()->where('site', $site->handle())->whereStatus('published')->count()) {
+                $sitemap .= $this->addSitemap('sitemap_statamic_collection_' . $site->handle() . '_' . $collection->handle() . '.xml');
+                $this->generateCollectionSitemap($site, $collection);
+            }
+        }
+
+        $taxonomies = TaxonomyFacade::all()
+            ->filter(function ($taxonomy) use ($site) {
+                $terms = $taxonomy->queryTerms()
+                    ->where('site', $site->handle())
+                    ->get();
+
+                return $terms
+                    ->filter(function ($term) {
+                        return view()->exists($term->template());
```

Comment:
can't you check for a template in the query? It's a bit more efficient, also don't have to check the is not empty then

Reply:
See https://github.com/rapidez/statamic/pull/78#issuecomment-2399337106
Taxonomies are limited in the way we can check for urls, Statamic seems to be taking this approach themselves as well.

Reply:
That's not what i'm saying, you can skip the whole `->filter()` if you doe a `->where` on the template in the query

Reply:
No thats not possible, the template is not something we can query on. You can see why here https://github.com/statamic/cms/blob/5.x/src/Taxonomies/LocalizedTerm.php#L381

Reply:
Was just able to test it, unfortunately an existing `uri` column doesn't guarantee an existing view. It seems like as long as a term has a slug, the `uri` field will be filled.

This is probably also why Statamic does it this way themselves in their SEO addon.

Reply:
Either way, still added the uri check in the query. Since all terms that do have views also do have uri's.